modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/defragmentation/IndexingDefragmentation.java

```diff
@@ -278,23 +287,14 @@ private boolean defragmentTable(
 
         int off = io.offset(idx);
 
-        IntMap<Value> values = new IntHashMap<>();
-
-        if (inlineIdxs != null) {
-            int fieldOff = 0;
+        int payloadSize = io.getPayloadSize();
 
-            for (int i = 0; i < inlineIdxs.size(); i++) {
-                AbstractInlineIndexColumn inlineIndexColumn = (AbstractInlineIndexColumn) inlineIdxs.get(i);
+        byte[] values;
 
-                Value value = inlineIndexColumn.get(pageAddr, off + fieldOff, io.getPayloadSize() - fieldOff);
-
-                fieldOff += inlineIndexColumn.inlineSizeOf(value);
-
-                final int columnIndex = inlineIndexColumn.columnIndex();
-
-                values.put(columnIndex, value);
-            }
-        }
+        if (inlineIdxs != null)
+            values = PageUtils.getBytes(pageAddr, off, payloadSize);
+        else
+            values = Utils.EMPTY_BYTES;
```

Comment:
I see that in IO you write `assert inlineIdxs != null : "no inline index helpers";`, but here it's a valid `else` branch. Which code do I trust?

Reply:
yes, you're right, that check is correct, and the other one is odd